scripts/analysis.py

```diff
@@ -904,65 +913,96 @@ def commod_per_inst(cur, commodity, timestep=10000):
             '" and time < ' + str(timestep)
-        inst_output_dict[inst_name] = cur.execute(query).fetchone()[0]
+        institution_output[inst_name] = cur.execute(query).fetchone()[0]
 
-    return inst_output_dict
+    return institution_output
 
 
-def get_waste_dict(isotope_list, mass_list, time_list, duration):
+def wastes(isotopes, mass_timeseries, duration):
     """Given an isotope, mass and time list, creates a dictionary
        With key as isotope and time series of the isotope mass.
 
-    Parameters
-    ----------
-    isotope_list: list
+    Inputs:
+    isotopes: list
         list with all the isotopes from resources table
-    mass_list: list
-        list with all the mass values from resources table
-    time_list: list
-        list with all the time values from resources table
-    duration: int
+    mass_timeseries: list
+        a list of lists.  each outer list corresponds to a different isotope
+        and contains tuples in the form (time,mass) for the isotope transaction.
+    duration: integer
         simulation duration
 
-    Returns
-    -------
-    waste_dict: dictionary
+    Outputs:
+    waste: dictionary
         dictionary with "key=isotope, and
         value=mass timeseries of each unique isotope"
     """
-    waste_dict = collections.OrderedDict()
-    isotope_set = set(isotope_list)
-    for iso in isotope_set:
-        mass = 0
-        time_mass = []
-        # at each timestep,
-        for i in range(0, duration):
-            # for each element in database,
-            for x, y in enumerate(isotope_list):
-                if i == time_list[x] and y == iso:
-                    mass += mass_list[x]
-            time_mass.append(mass)
-        waste_dict[iso] = time_mass
 
-    return waste_dict
+    keys = []
+    for key in isotopes:
+        keys.append(key)
+
+    waste = {}
+
+    if len(mass_timeseries) == 1:
+        times = []
+        masses = []
+        for i in list(mass_timeseries[0]):
+            time = str(i).split(',')[0]
+            times.append((float(time.strip('('))))
+            mass = str(i).split(',')[1]
+            masses.append((float(mass.strip(')').strip('('))))
+
+        time_developed = times
+        masses1 = masses
+        nums = np.arange(0, duration)
+
+        for j in nums:
+            if j not in time_developed:
+                time_developed.insert(j, j)
+                masses1.insert(j, 0)
+
+        waste[key] = masses1
+
+    else:
+        for element in range(len(mass_timeseries)):
+            times = []
+            masses = []
+            for i in list(mass_timeseries[element]):
+                time = str(i).split(',')[0]
+                times.append((float(time.strip('('))))
+                mass = str(i).split(',')[1]
+                masses.append((float(mass.strip(')').strip('('))))
+
+            time_developed = times
+            masses1 = masses
+            nums = np.arange(0, duration)
+
+            for j in nums:
```

Comment:
It seems this whole thing is copy pasted from above -- why ? What is different about this else loop? Probably this should be it own function. The DRY principle of programming is "do not repeat yourself".